LABTASKS.md

```diff
@@ -0,0 +1,90 @@
+- When displaying the ToDos in your Angular front-end, make thoughtful decisions about whether work like filtering
+  should be done in Angular or via database queries. For example, have the database filter out all the ToDos belonging to a single user,
+  but let Angular filter by category, body, or status.
+- **Do at least some filtering on the database side of things.**
+
+Features to implement:
+
+- Filter Todos by status
+- Filter Todos by owner
+- Filter Todos by contents of the body
+- Filter Todos by category
+- Ability to add new Todos. This requires:
+  - An Angular form/page that allows the user to enter the information for a new todo with reasonable controls and validation.
+  - A new endpoint on the server that allows clients to add new Todos.
+  - Logic in the server to add that new Todo to the database and return the new ID.
+- Testing of all of the above.
+
+## Workflow and Continuous Integration
+
+You should organize your work into features, where each feature is an _epic_ in ZenHub. Each feature should "slice the cake", including server implementation and testing, Angular implementation and testing, and end-to-end testing.
+
+:warning: Keep the features as small and focussed as
+possible. Otherwise things like code reviews become a huge burden,
+and pull requests become far less useful.
+
+To maximize your learning, we strongly encourage you to implement features in Angular one at a time instead of copying over
+entire components from the previous lab that presumably support numerous features.
+
+- As you work, create a branch for a new feature.
+  - Write tests for the server actions for the feature you added (run in the server folder using `./gradlew test` or `./gradlew test jacocoTestReport` as described in [README.md](./README.md#testing-the-server)). Then write the server code that makes those tests pass.
+  - Write new end-to-end tests for the new views using Cypress. Run
+    these in client folder using `ng e2e --watch` as described in [README.md](./README.md#end-to-end-testing)).
+  - Write unit tests for the new Angular components you are adding using Karma (run in client folder using `ng test` or `ng test --code-coverage` as described in [README.md](./README.md#testing-the-client)). Then write the Angular code that makes those tests pass.
+  - Address failing builds.
+- Perform code reviews especially if you're not pair programming.
+  _Ask (your partner, the class, the instructor) about things you don't understand!_
+- Use pull requests or work together with your lab partner to
+  merge things into master when a feature is working
+  and is tested (with passing tests and decent coverage).
+
+## Writing Todos to the Database
+
+- We have included an example of writing to the database with `addUser` functionality. Add to both the front-end and back-end to make it possible to add ToDos so that they appear both in your list and in the database.
```

Comment:
```suggestion
- We have included an example of writing to the database with `addUser` functionality. Add to both the front-end and back-end to make it possible to add todos so that they appear both in your list and in the database.
```